src/binder/bind/bind_ddl.cpp

```diff
@@ -113,9 +113,10 @@ uint32_t Binder::bindPrimaryKey(
     return primaryKeyIdx;
 }
 
-property_id_t Binder::bindPropertyName(TableSchema* tableSchema, string propertyName) {
+property_id_t Binder::bindPropertyName(TableSchema* tableSchema, const string& propertyName) {
     for (auto& property : tableSchema->properties) {
         if (property.name == propertyName) {
+            validateNotPrimaryKey(tableSchema, property.propertyID);
```

Comment:
bindPropertyName bind a string name to internal property id. But should not (at least from the function name) validate whether or not the property is primary key or not.

`validateNotPrimaryKey` this function seems to be very specific to drop property statement. I would move the logic there and NOT create a function.

Reply:
Deleted the function: `validateNotPrimaryKey`, and moved the check inside `bindDropProperty`